Makefile

```diff
@@ -57,7 +57,7 @@ ebin/gradualizer.app: src/gradualizer.app.src | ebin
 	cp $< $@
 
 .PHONY: shell
-shell:
+shell: $(beams)
```

Comment:
Maybe also `ebin/gradualizer.app` should be a prerequisite? I think it may be needed by `application` which is started on demand within the code.

Alternatively, just add `app` as a prerequisite.

Reply:
I thought it wasn't but you're most likely right